src/App.tsx

```diff
@@ -1,26 +1,244 @@
+import { ErrorType } from './types/ErrorType';
+
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.ERROR_DEFAULT,
+  );
+  const [filterTodoBy, setFilterTodoBy] = useState<FilterType>(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingByIds, setLoadingByIds] = useState<number[]>([]);
+  const isAllCompletedTodo = todos.every(todo => todo.completed);
+  const isVisibileBtn = todos.length > 0;
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const asyncFetch = async () => {
+      inputRef.current?.focus();
+
+      try {
+        const loadTodos = await todoServices.getTodos();
+
+        setTodos(loadTodos);
+      } catch (error) {
+        setErrorMessage(ErrorType.ERROR_LOADING);
+        throw error;
+      }
+    };
+
+    asyncFetch();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      if (FilterType.ACTIVE === filterTodoBy) {
+        return !todo.completed;
+      }
+
+      if (FilterType.COMPLETED === filterTodoBy) {
+        return todo.completed;
+      }
+
+      return true;
+    });
+  }, [todos, filterTodoBy]);
+
+  const addTodo = ({ id, userId, title, completed }: Todo) => {
+    setErrorMessage(ErrorType.ERROR_DEFAULT);
+    setIsLoading(true);
+
+    const newTempTodo = { id, userId, title, completed };
+
+    setTempTodo(newTempTodo);
+
+    return todoServices
+      .createTodo({ title, userId, completed })
+      .then(newTodo => setTodos(currentTodo => [...currentTodo, newTodo]))
+      .catch(error => {
+        setErrorMessage(ErrorType.ERROR_ADD);
+
+        throw error;
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setErrorMessage(ErrorType.ERROR_DEFAULT);
+    setLoadingByIds(prev => [...prev, todoId]);
+
+    return todoServices
+      .deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodo => currentTodo.filter(todo => todo.id !== todoId));
+      })
+      .catch(error => {
+        setErrorMessage(ErrorType.ERROR_DELETE);
+
+        throw error;
+      })
+      .finally(() => {
+        inputRef.current?.focus();
+        setLoadingByIds(prev => prev.filter(id => id !== todoId));
+      });
+  };
+
+  const clearCompleted = () => {
+    const completedTodoId = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    const deletionCompletedTodo = completedTodoId.map(id => deleteTodo(id));
+
+    Promise.all(deletionCompletedTodo);
```

Comment:
Promise.all possible reason of failed tests.
use forEach method - responses should be resolved independetly each other.